prettier/main.ts

```diff
@@ -317,6 +333,18 @@ async function main(opts): Promise<void> {
     write: opts["write"]
   };
 
+  const tty = Deno.isTTY();
+
+  const shouldReadFromStdin =
+    (!tty.stdin && (tty.stdout || tty.stderr)) || !!opts["from-stdin"];
```

Comment:
Q: Why add `from-stdin`?

A:
Because in the pipeline, you can't use `isTty` to judge when you need to read from stdin
So in the test case, the display specifies to read from stdin